var/spack/repos/builtin/packages/tangram/package.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Tangram(CMakePackage):
+    """Tangram is an material interface reconstruction package used in
+    multimaterial ALE codes and multi-material remapping
+    (https://github.com/laristra/portage)
+    """
+
+    homepage = "https://portage.lanl.gov"
+    git      = "https://github.com/laristra/tangram.git"
+    url      = "https://github.com/laristra/tangram/releases/download/1.0.1/tangram-1.0.1.tar.gz"
+
+    maintainers = ['raovgarimella']
+
+    version('1.0.1', sha256='8f2f8c01bb2d726b0f64e5a5bc3aa2bd8057ccaee7a29c68f1439d16e39aaa90')
+    version('master', branch='master', submodules=True)
+
+    variant('mpi', default=True, description='Enable interface reconstruction on distributed meshes with MPI')
+    variant('thrust', default=False, description='Enable on-node parallelism using NVidia Thrust library')
+    variant('kokkos', default=False, description='Enable on-node or device parallelism with Kokkos')
+    variant('openmp', default=False, description="Enable on-node parallelism using OpenMP")
+    variant('cuda', default=False, description="Enable GPU parallelism using CUDA")
+
+    # wrappers to enable external mesh/state libraries (needed only for testing)
+    variant('jali', default=False, description='Build with Jali mesh infrastructure (for testing)')
```

Comment:
I don't see where these last 3 variants are used in this package. If users want to control how dependencies are built, they can already do this:
```console
$ spack install tangram ^wonton+jali~cuda
```

Reply:
It is my attempt at variant forwarding. Without it, folks will have to drill down into Wonton to see what is available and what is not. But not all of the dependent packages variants may be applicable to the primary package.

Reply:
@adamjstewart Just checking if your comment was a suggestion or a required change per Spack guidelines. If it is a requirement, I will do what is necessary to get the PR approved. Thanks

Reply:
Not a hard requirement, I'm just trying to limit how much variant forwarding we do. Some packages have gotten out of hand, with dozens of variants that don't affect the package at all.

> Without it, folks will have to drill down into Wonton to see what is available and what is not.

Folks currently have to drill down into the Tangram package to see what is available and what is not. Why not keep the complexity in a single location (Wonton) so that users can set things once in `packages.yaml` and forget about it?

Reply:
Adam, actually, I missed out using the Jali variant. It does go in as a CMake option. The only things that are forwarded then are the 'cuda' and the 'openmp' because they are both off by default and get turned on only for `kokkos` or `thrust`. I don't anticipate many more variants being added. This package (and its counterparts Portage and Wonton) are becoming somewhat mature.

Reply:
@adamjstewart Can you re-review? Right now only two variants are forwarded. If even these are not acceptable, I will have to go back to Wonton and introduce more defaults and checks. Thank you